src/Core/Geographic/Crs.ts

```diff
@@ -80,100 +85,92 @@ function toUnitWithError(crs) {
 }
 
 /**
- * This module provides basic methods to manipulate a CRS (as a string).
- *
- * @module CRS
+ * This module provides basic methods to manipulate a CRS.
  */
 export default {
     /**
      * Units that can be used for a CRS.
-     *
-     * @enum {number}
      */
     UNIT,
 
     /**
-     * Assert that the CRS is valid one.
+     * Assert that the CRS is a valid one.
      *
-     * @param {string} crs - The CRS to validate.
+     * @param crs - The CRS to validate.
      *
-     * @throws {Error} if the CRS is not valid.
+     * @throws {@link Error} if the CRS is not valid.
      */
-    isValid(crs) {
+    isValid(crs: ProjectionLike) {
         toUnitWithError(crs);
     },
 
     /**
      * Assert that the CRS is geographic.
      *
-     * @param {string} crs - The CRS to validate.
-     * @return {boolean}
-     * @throws {Error} if the CRS is not valid.
+     * @param crs - The CRS to validate.
+     * @throws {@link Error} if the CRS is not valid.
      */
-    isGeographic(crs) {
+    isGeographic(crs: ProjectionLike) {
         return (toUnitWithError(crs) == UNIT.DEGREE);
     },
 
     /**
      * Assert that the CRS is using metric units.
      *
-     * @param {string} crs - The CRS to validate.
-     * @return {boolean}
-     * @throws {Error} if the CRS is not valid.
+     * @param crs - The CRS to validate.
+     * @throws {@link Error} if the CRS is not valid.
      */
-    isMetricUnit(crs) {
+    isMetricUnit(crs: ProjectionLike) {
         return (toUnit(crs) == UNIT.METER);
     },
 
     /**
      * Get the unit to use with the CRS.
      *
-     * @param {string} crs - The CRS to get the unit from.
-     * @return {number} Either `UNIT.METER`, `UNIT.DEGREE` or `undefined`.
+     * @param crs - The CRS to get the unit from.
+     * @returns Either `UNIT.METER`, `UNIT.DEGREE` or `undefined`.
      */
     toUnit,
 
     /**
-     * Is the CRS EPSG:4326 ?
+     * Is the CRS EPSG:4326?
      *
-     * @param {string} crs - The CRS to test.
-     * @return {boolean}
+     * @param crs - The CRS to test.
      */
     is4326,
     /**
-     * Is the CRS geocentric ?
+     * Is the CRS geocentric?
      * if crs isn't defined the method returns false.
      *
-     * @param {string} crs - The CRS to test.
-     * @return {boolean}
+     * @param crs - The CRS to test.
      */
     isGeocentric,
```

Comment:
Not related to pure TS migration but on a code cleanliness consideration, we could define this the same way we export `isGeographic` or `reasonnableEpsilon` : we could either define them all outside of the `export DEFAULT` object and reference them here, or define them all directly in the `export DEFAULT` object.

Reply:
Better, in my opinion, we should just not use `export default` anymore in the codebase. There are not compatible with tree-shaking. Imagine we build a bundle importing `Crs.js` and just using the `isGeocentric`, with `export default` the whole code of `Crs.js` is bundled, with multiple `export` only this function is bundled.

Reply:
However, this may be outside the scope of this PR?

Reply:
thanks :)